control/controlclient/direct.go

```diff
@@ -200,6 +206,17 @@ func NewDirect(opts Options) (*Direct, error) {
 	return c, nil
 }
 
+// Close closes any open idle connections.
+// TODO(maisem): make this do more.
+func (c *Direct) Close() error {
+	c.mu.Lock()
+	defer c.mu.Unlock()
+	if c.noiseClient != nil {
+		c.noiseClient.CloseIdleConnections()
```

Comment:
if we're closing the connection I think we want to be more aggressive than this and actually close all the underlying TCP connection(s) immediately

Reply:
I don't follow, wdym? Are you saying that we don't keep any underlying TCP connection open at all?

Reply:
Done